### PR TITLE
no need to set ClusterIP when syncing headless service

### PR DESCRIPTION
**Pull request by LinuxGit**
### What problem does this PR solve? 

remove set ClusterIP operation when syncing headless service

### Check List 

Tests 

 - No code

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

**Comment by LinuxGit** [comment]
I added a commit to remove op in tidb_member_manager.go and dismissed your review.
PTAL again. @weekface

File: pkg/manager/member/pd_member_manager.go [hunk 1]
```diff
@@ -152,8 +152,6 @@ func (pmm *pdMemberManager) syncPDHeadlessServiceForTidbCluster(tc *v1alpha1.Tid
 	if !equal {
 		svc := *oldSvc
 		svc.Spec = newSvc.Spec
-		// TODO add unit test
-		svc.Spec.ClusterIP = oldSvc.Spec.ClusterIP
```
Comment (xiaojingchen) [inline pkg/manager/member/pd_member_manager.go]: this line is to ensure that the `ClusterIP` remains unchanged when svc updated.

File: pkg/manager/member/tidb_member_manager.go [hunk 2]
```diff
@@ -115,8 +115,6 @@ func (tmm *tidbMemberManager) syncTiDBHeadlessServiceForTidbCluster(tc *v1alpha1
 	if !equal {
 		svc := *oldSvc
 		svc.Spec = newSvc.Spec
-		// TODO add unit test
-		svc.Spec.ClusterIP = oldSvc.Spec.ClusterIP
```
Comment (xiaojingchen) [inline pkg/manager/member/tidb_member_manager.go]: ditto

File: pkg/manager/member/pd_member_manager.go (hunk 1 above)
Comment (LinuxGit) [reply pkg/manager/member/pd_member_manager.go]: Yes. syncPDServiceForTidbCluster should keep ClusterIP unchanged. But when syncPDHeadlessServiceForTidbCluster, there's no need to set it, it should be default value of  'None' from newSvc.Spec.

Comment (xiaojingchen) [reply pkg/manager/member/pd_member_manager.go]: ok, that's right.

**Pull request merged**